### PR TITLE
[improvement] stream load data is converted to json format

### LINKED ISSUE #14: [improvement] stream load import only support json data
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

3.1_2.12-1.0.1

### What's Wrong?

If the data field contains \n  \t separator, too many filtered rows will appear when importing data

### What You Expected?

When importing data, the data is assembled into JSON format, and stream load is imported in JSON format .

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by smallhibiscus**
# Proposed changes

Issue Number: close #14

## Problem Summary:

Stream load data is converted to json format.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java [hunk 1]
```diff
@@ -159,6 +160,25 @@ public void load(List<List<Object>> rows) throws StreamLoadException {
...
+                    }
+                }
+                dataList.add(dataMap);
+            }
+        } catch (Exception e) {
+            throw new StreamLoadException("The number of configured columns does not match the number of data columns.");
+        }
+        load((new ObjectMapper()).writeValueAsString(dataList));
+    }
+
```
Comment (JNSimba) [inline spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java:181]: It is recommended to add a unit test.

File: spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java [hunk 2]
```diff
@@ -35,13 +38,7 @@
...
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Base64;
-import java.util.Calendar;
-import java.util.List;
-import java.util.StringJoiner;
-import java.util.UUID;
+import java.util.*;
```
Comment (JNSimba) [inline spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java:41]: not import *

File: spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java (hunk 1 above)
Comment (smallhibiscus) [reply spark-doris-connector/src/main/java/org/apache/doris/spark/DorisStreamLoad.java:181]: OK,I will add unit test.

**Review by hf200012 — APPROVED** [review]
LGTM

File: spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala [hunk 3]
```diff
@@ -93,7 +93,7 @@ private[sql] class DorisSourceProvider extends DataSourceRegister
 
           for (i <- 1 to maxRetryTimes) {
             try {
-              dorisStreamLoader.load(rowsBuffer)
```
[the remark this answers is not in the document]
Comment (smallhibiscus) [reply spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala:96]: Thx,I will delete it.

File: spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala [hunk 4]
```diff
@@ -60,7 +60,7 @@ private[sql] class DorisSourceProvider extends DataSourceRegister
     val sparkSettings = new SparkSettings(sqlContext.sparkContext.getConf)
     sparkSettings.merge(Utils.params(parameters, logger).asJava)
     // init stream loader
-    val dorisStreamLoader = new DorisStreamLoad(sparkSettings)
+    val dorisStreamLoader = new DorisStreamLoad(sparkSettings,data.columns)
```
Comment (JNSimba) [inline spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala:63]: format code

Comment (smallhibiscus) [reply spark-doris-connector/src/main/scala/org/apache/doris/spark/sql/DorisSourceProvider.scala:63]: OK,thx

**Comment by JNSimba** [comment]
LGTM

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**